tests/unit/unit1614.c

```diff
@@ -77,6 +77,12 @@ UNITTEST_START
     { NULL, NULL, 0, FALSE} /* end marker */
   };
   struct noproxy list[]= {
+    { "www.example.com", "localhost,.example.com,.example.de", TRUE},
+    { "www.example.com.", "localhost,.example.com,.example.de", TRUE},
+    { "example.com", "localhost,.example.com,.example.de", TRUE},
+    { "example.com.", "localhost,.example.com,.example.de", TRUE},
+    { "www.example.com", "localhost,.example.com.,.example.de", TRUE},
+    { "www.example.com", "localhost,www.example.com.,.example.de", TRUE},
```

Comment:
Maybe also test for `example.com` and not just `.example.com`, some tools out there can not do the leading dot so people might not have it in their env.

Not too related to the issue at hand, but for sure a good improvement of the test-suite to prevent potential problems in the future.

Reply:
A good idea. I added three more checks. That what you had in mind?

Reply:
Yes, looking good!